c-sharp-razor/src/Minitwit.Infrastructure/Repository/AuthorRepository.cs

```diff
@@ -0,0 +1,269 @@
+            throw new Exception("This author has no Messages");
+
+        if (page < 1)
+            page = 1;
+
+        int pageSizeIndex = (page - 1) * PageSize;
+
+        if (Messages.Count < pageSizeIndex + PageSize)
+            return Messages
+                .ToList()
+                .GetRange(pageSizeIndex, Messages.Count - pageSizeIndex)
+                .OrderByDescending(c => c.TimeStamp)
+                .ToList();
+        if (Messages.Count > PageSize)
+            return Messages
+                .ToList()
+                .GetRange(pageSizeIndex, PageSize)
+                .OrderByDescending(c => c.TimeStamp)
+                .ToList();
+        return Messages.OrderByDescending(c => c.TimeStamp).ToList();
+    }
+
+    public async Task<ICollection<Message>> GetMessagesByAuthorAndFollowingAsync(int id)
+    {
+        ICollection<Message> Messages = new List<Message>(await GetMessagesByAuthorAsync(id));
+
+        foreach (Author author in await GetFollowingByIdAsync(id))
+        {
+            Messages = Messages.Concat(await GetMessagesByAuthorAsync(author.Id)).ToList();
+        }
+
+        return Messages;
+    }
+
+    public async Task<ICollection<Message>> GetMessagesByAuthorAndFollowing(int id, int page)
+    {
+        Author? author = await GetAuthorByIdAsync(id);
+        //Get Messages from the author, and append Messages from followers to that list
+        ICollection<Author> following = await GetFollowingByIdAsync(id);
+        ICollection<Message> Messages = new List<Message>();
+
+        // Add all the users Messages to the list without pagination
+        foreach (var MessageDto in await GetMessagesByAuthorAsync(id))
+            Messages.Add(MessageDto);
+
+        foreach (Author? follower in following)
+        {
+            ICollection<Message> followingMessages = await GetMessagesByAuthorAsync(follower.Id);
+            //If follower has no Messages, skip them
+            if (followingMessages.Count == 0)
+            {
+                continue;
+            }
+
+            //Add each Message from the follower to the list
+            //TODO Try to find alternative to foreach
+            foreach (var MessageDto in followingMessages)
+            {
+                Messages.Add(MessageDto);
+            }
+        }
+        //Sort the Messages according to timestamp, latest first
+        Messages = Messages.OrderByDescending(c => c.TimeStamp).ToList();
+
+        int pageSizeIndex = (page - 1) * PageSize;
+
+        if (Messages.Count < pageSizeIndex + PageSize)
+            return Messages.ToList<Message>().GetRange(pageSizeIndex, Messages.Count - pageSizeIndex);
+        if (Messages.Count > PageSize)
+            return Messages.ToList<Message>().GetRange(pageSizeIndex, PageSize);
+        return Messages;
+    }
+
+    // ----- Get Messages By Author Methods ----- //
+    public async Task<int> GetMessageCountByAuthorAsync(int authorId)
```

Comment:
This comment here is a general comment for all this service. They are putting a lot of Async in the method name and I think it is not cool since it has already the async in the signature...